openfe/tests/setup/test_openmm_equil_protocols.py

```diff
@@ -751,3 +753,20 @@ def test_constraints(tyk2_xml, tyk2_reference_xml):
             assert a.get('p1') == b.get('p1')
             assert a.get('p2') == b.get('p2')
             assert float(a.get('d')) == pytest.approx(float(b.get('d')))
+
+
+def test_custom_openmm_unit_codec():
```

Comment:
Note that this test doesn't check the case that you're starting analysis without having imported this `Protocol`. There's a potential issue there. We might dodge it in this specific case, but it is likely to be a problem in a more general case.

Possible workaround proposed here: https://github.com/OpenFreeEnergy/gufe/pull/68#issuecomment-1267441746